### PR TITLE
Add button to duplicate pages

### PR DESCRIPTION
**Pull request by der-On**
This PR adds a "duplicate" button to each static page. With it a static page can be easily duplicated.

We use it to create "template" pages for site editors that then simply can be duplicated and edited.
Especially with sites containing repeaters it is not possible to copy paste content from one page to another and having a starting point always helps site editors.

### CODE REVIEW

**Comment by LukeTowers** [comment]
Do you have a screenshot or screen recording of this in action?

File: assets/js/pages-page.js
Comment (bennothommo) [inline assets/js/pages-page.js:384]: @der-On unless I'm mistaken, this will always be `null`. There doesn't seem to be a parent defined in the data for the duplicate button. I don't think this is needed anyway, as the parent will be copied from the original page, so you could probably remove this and any references to parent in here and the PHP handler.

File: controllers/Index.php
Comment (bennothommo) [inline controllers/Index.php:203]: @der-On Please use `===` for string comparisons.

File: controllers/Index.php
Comment (bennothommo) [inline controllers/Index.php:226]: You should probably retrieve these variables from the original page.

**Review by bennothommo — DISMISSED** [review]

File: controllers/Index.php
Comment (der-On) [reply controllers/Index.php:203]: This was copy pasted from `onCreateObject` ;) but sure I will.

File: assets/js/pages-page.js
Comment (der-On) [reply assets/js/pages-page.js:384]: Yes, I've discovered right after making the PR that duplicating a nested page places it at the root level. Will adjust the PR to include the parent so nested pages get duplicated into the parent.

**Comment by der-On** [comment]
> Do you have a screenshot or screen recording of this in action?

I'll do some screenshots.

**Comment by der-On** [comment]
@LukeTowers here you see the new duplicate button and what happens when you click it:
Bildschirmfoto am 2022-03-15 um 08 44 31-fullpage
Bildschirmfoto am 2022-03-15 um 08 44 43-fullpage

**Comment by der-On** [comment]
I've also added a label to the duplicate button as the icon is not really self explanatory. I hope I've got all translations correct.

File: lang/nl/lang.php
Comment (AIC-BV) [inline lang/nl/lang.php:89]: Something minor;
In Dutch, the verb 'To duplicate' is 'Dupliceren'
'Duplicaat' is a noun, a duplicated item

This is probably the case for the other languages as well
I guess you used google translate 'Duplicate' but you should've used 'To duplicate' instead

**Comment by LukeTowers** [comment]
@der-On please revert the changes to translations for languages that you don't actually speak. Generally it's better if people see the fallback (English) rather than a google translated version which could be wildly incorrect.

**Comment by der-On** [comment]
> @der-On please revert the changes to translations for languages that you don't actually speak. Generally it's better if people see the fallback (English) rather than a google translated version which could be wildly incorrect.

Absolutely right. I will do so.

File: lang/nl/lang.php
Comment (der-On) [reply lang/nl/lang.php:89]: I've actually translated from german "duplizieren" wich is a verb. I will fix it and remove for all languages I do not speak.

**Comment by LukeTowers** [comment]
Thanks @der-On! Looks good to me, will wait on @bennothommo for final review of this.

**Comment by bennothommo** [comment]
Last change that I would request is to remove the label "Duplicate" next to the icon for the duplicate button, and use a tooltip instead, like the "delete" button next to it. Duplicating a page is not a primary action, so I think it would be better to minimise it's UI footprint.

**Comment by LukeTowers** [comment]
Hmm, I don't personally mind it having the label there. It's not like it's taking up too much space, and the icon by itself can be pretty hard to see / understand what it's there for. If anything having the label there improves the context clues that the delete button and the calendar icon have purpose.

Thoughts?

**Comment by bennothommo** [comment]
It's a subjective topic, but since you'll likely be using the duplicate button much less than the save or preview buttons, I don't think it's necessarily important enough to be prominent. Perhaps the icon can be changed to better reflect its use?

**Comment by LukeTowers** [comment]
The icon is probably fine, can't think of a better one off the top of my head. Unless anyone has strong arguments in favour of keeping the label instead of turning it into a tooltip for display on hover over I'm fine with either approach.

**Comment by der-On** [comment]
> The icon is probably fine, can't think of a better one off the top of my head. Unless anyone has strong arguments in favour of keeping the label instead of turning it into a tooltip for display on hover over I'm fine with either approach.

Completely fine with that. I will turn it into a tooltip. It's a secondary action and thus should use less space.

**Comment by der-On** [comment]
Is there anything holding this back?

**Review by LukeTowers — APPROVED** [review]

**Comment by LukeTowers** [comment]
@bennothommo will leave for you to merge for handling the release

**Pull request merged**
